### PR TITLE
feat(js): adding vertex retriever & google search options

### LINKED ISSUE #547: Support Vertex AI Gemini grounding with Google Search and Vertex data stores
**Is your feature request related to a problem? Please describe.**

Vertex AI offers the ability to [ground responses for Gemini models](https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/ground-gemini) using Google Search or your own data in within Vertex AI Search datas stores.

It appears to supported in the Python and [Node.js SDK](https://cloud.google.com/vertex-ai/generative-ai/docs/reference/nodejs/latest#grounding-using-vertex-ai-search-preview), but not yet in the Go SDK. 

In the Vertex AI Gemini API, it works by passing in the grounding mechanism as a tool to the model, like so (Node.js):

```JavaScript
async function generateContentWithVertexAISearchGrounding() {
  const generativeModelPreview = vertexAI.preview.getGenerativeModel({
    model: textModel,
    // The following parameters are optional
    // They can also be passed to individual content generation requests
    safetySettings: [{category: HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT, threshold: HarmBlockThreshold.BLOCK_MEDIUM_AND_ABOVE}],
    generationConfig: {maxOutputTokens: 256},
  });

  const vertexAIRetrievalTool = {
    retrieval: {
      vertexAiSearch: {
        datastore: 'projects/.../locations/.../collections/.../dataStores/...',
      },
      disableAttribution: false,
    },
  };
  const result = await generativeModelPreview.generateContent({
    contents: [{role: 'user', parts: [{text: 'Why is the sky blue?'}]}],
    tools: [vertexAIRetrievalTool],
  })
  const response = result.response;
  const groundingMetadata = response.candidates[0].groundingMetadata;
  console.log("Grounding metadata is: ", JSON.stringify(groundingMetadata));
}
generateContentWithVertexAISearchGrounding();
```

I'm not sure how this would work with Genkit's current design for specifying tools in a generation request.

### PR DESCRIPTION
**Pull request by pr-Mais**
Resolves #547

Checklist (if applicable):
- [x] Tested (manually, unit tested, etc.)
- [ ] Changelog updated
- [ ] Docs updated

### CODE REVIEW

**Comment by cabljac** [comment]
@pr-Mais does Google AI support this on Gemini? or just Vertex

**Comment by pr-Mais** [comment]
> @pr-Mais does Google AI support this on Gemini? or just Vertex

Docs only mention Vertex API

File: js/plugins/vertexai/src/gemini.ts [hunk 1]
```diff
@@ -53,8 +53,21 @@ const SafetySettingsSchema = z.object({
   threshold: z.nativeEnum(HarmBlockThreshold),
 });
 
+const VertexRetrievalSchema = z.object({
+  datastore: z.object({
+    projectId: z.string(),
+    location: z.string(),
```
Comment (pavelgj) [inline js/plugins/vertexai/src/gemini.ts:59]: can you make these optional? can default to the project/location that we use for the vertex client: https://github.com/firebase/genkit/blob/main/js/plugins/vertexai/src/index.ts#L105-L106

Comment (pr-Mais) [reply js/plugins/vertexai/src/gemini.ts]: done

**Review by pavelgj — APPROVED** [review]

File: js/plugins/vertexai/src/gemini.ts [hunk 2]
```diff
@@ -47,14 +47,28 @@ import {
...
+const VertexRetrievalSchema = z.object({
+  datastore: z.object({
+    projectId: z.string().optional(),
+    location: z.string().optional(),
+    dataStoreId: z.string(),
+  }),
+  disableAttribution: z.boolean().optional(),
+});
+
+const GoogleSearchRetrievalSchema = z.object({});
```
Comment (ssbushi) [inline js/plugins/vertexai/src/gemini.ts:66]: GoogleSearchRetrieval options supports `disableAttribution`. https://github.com/googleapis/nodejs-vertexai?tab=readme-ov-file#grounding-preview

File: js/plugins/vertexai/src/gemini.ts [hunk 3]
```diff
@@ -508,6 +529,29 @@ export function geminiModel(name: string, vertex: VertexAI): ModelAction {
...
+        const _projectId =
+          vertexRetrieval.datastore.projectId || options.projectId;
+        const _location =
+          vertexRetrieval.datastore.location || options.location;
+        const _dataStoreId = vertexRetrieval.datastore.dataStoreId;
+        const datastore = `projects/${_projectId}/locations/${_location}/collections/default_collection/dataStores/${_dataStoreId}`;
+        chatRequest.tools?.push({
+          retrieval: {
+            vertexAiSearch: {
+              datastore: datastore,
```
Comment (ssbushi) [inline js/plugins/vertexai/src/gemini.ts:549]:
```suggestion
              datastore,
```

**Review by ssbushi — APPROVED** [review]

**Pull request merged**